Source/Evolve/WarpXEvolve.cpp

```diff
@@ -398,19 +398,43 @@ WarpX::OneStep_nosub (Real cur_time)
 
     ExecutePythonCallback("afterdeposition");
 
-    // Synchronize J and rho: filter, exchange boundary, interpolate across levels.
-    // With Vay current deposition, the current deposited at this point is not yet
-    // the actual current J. This is computed later in WarpX::PushPSATD, by calling
-    // WarpX::PSATDVayDeposition. The function SyncCurrent is called after that,
-    // instead of here, so that we synchronize the correct current.
-    // With current centering, the nodal current is deposited in 'current_fp_nodal':
-    // SyncCurrent stores the result of its centering into 'current_fp' and then
-    // performs both filtering, if used, and exchange of guard cells.
-    if (WarpX::current_deposition_algo != CurrentDepositionAlgo::Vay)
+    // Synchronize J and rho:
+    // filter (if used), exchange guard cells, interpolate across MR levels
+    if (maxwell_solver_id == MaxwellSolverAlgo::PSATD)
+    {
+        if (fft_periodic_single_box)
+        {
+            // With periodic single box, synchronize J and rho here,
+            // even with current correction or Vay deposition
+            if (current_deposition_algo == CurrentDepositionAlgo::Vay)
+            {
+                SyncCurrent(current_fp_vay, current_cp);
```

Comment:
Should `current_cp` be replaced by `current_cp_vay` in theory?

Reply:
In theory yes, but I think we do not have the `current_cp_vay` MultiFab defined anywhere yet, because Vay deposition is not currently fully implemented with mesh refinement. Maybe I could add a note as inline comment ("TODO") to remember that this will need to be changed once Vay deposition is fully implemented with mesh refinement.

Reply:
Inline comment added in https://github.com/ECP-WarpX/WarpX/pull/3290/commits/7fcd631358aa22a57f8e3f63e93b8450a9ee99d0.